server/routes/receiveData/index.js

```diff
@@ -1,10 +1,467 @@
+
+var CurrLionCurrent = require('./../../database/models/currentData/lion/currLionCurrent.js')
+var CurrLionDischarge = require('./../../database/models/currentData/lion/currLionDischarge.js')
+var CurrLionVoltage = require('./../../database/models/currentData/lion/currLionVoltage.js')
+
+var CurrRadioTemperature = require('./../../database/models/currentData/radio/currRadioTemperature.js')
+var CurrRadioVoltage = require('./../../database/models/currentData/radio/currRadioVoltage.js')
+
+var CurrDigitalOutput = require('./../../database/models/currentData/currDigitalOutput.js')
+var CurrIrReading = require('./../../database/models/currentData/currIrReading.js')
+var CurrPhotoDiode = require('./../../database/models/currentData/currPhotoDiode.js')
+
+var AttitudeImuAccelerometer = require('./../../database/models/dataAttitude/attitudeImuAccelerometer.js')
+var AttitudeImuGyroscope = require('./../../database/models/dataAttitude/attitudeImuGyroscope.js')
+var AttitudeImuMagnetometer = require('./../../database/models/dataAttitude/attitudeImuMagnetometer.js')
+var AttitudeIrSensor = require('./../../database/models/dataAttitude/attitudeIrSensor.js')
+var AttitudeMagnetometer = require('./../../database/models/dataAttitude/attitudeMagnetometer.js')
+var AttitudePhotoDiode = require('./../../database/models/dataAttitude/attitudePhotoDiode.js')
+
+var FlashBurstLifepoCurrent = require('./../../database/models/dataFlashBurst/flashBurstLifepoCurrent.js')
+var FlashBurstLifepoVoltage = require('./../../database/models/dataFlashBurst/flashBurstLifepoVoltage.js')
+var FlashBurstTemperature = require('./../../database/models/dataFlashBurst/flashBurstTemperature.js')
+
+var RadioTransmissionCurrent = require('./../../database/models/dataRadioTransmission/radioTransmissionCurrent.js')
+var RadioTransmissionLionVoltage = require('./../../database/models/dataRadioTransmission/radioTransmissionLionVoltage.js')
+var RadioTransmissionTemperature = require('./../../database/models/dataRadioTransmission/radioTransmissionTemperature.js')
+
+var promises = []
+
 // Main route for receiving data
 router.post('/', function(req, res, next){
+	promises = []
+	saveAll(req.body, res)
+});
 
+function saveAll(rawJSON, res) {
+	saveRaw(rawJSON)
+	.then(function(obj){
+		savePreamble(rawJSON.preamble, obj.tid)
+		saveErrorInfo(rawJSON.error_info, obj.tid)
+		saveCurrentData(rawJSON.current_data_info, obj.tid)
+		saveAttitudeData(rawJSON.data_attitude, obj.tid)
+		saveDataFlashBurst(rawJSON.data_flash_burst, obj.tid)
+		saveDataRadioTransmission(rawJSON.data_radio_transmission, obj.tid)
+		return Promise.all(promises)
+	})
+	.then(function(){
+		return res.sendStatus(200)
+	})
+	.catch(function(error){
+		console.error(error)
+		return res.sendStatus(500)
+	});
+}
 
-});
+function saveRaw(rawJSON) {
+	return (Raw.build({
+		raw: JSON.stringify(rawJSON)
+	}).save())
+}
+
+function savePreamble(preamble, tid) {
+	promises.push(Preamble.build({
```

Comment:
so this should return the Preamble.build promise rather than push it into the global promises array. instead, in saveAll, the promise array should be defined there and pushed in there